controllers/registry.go

```diff
@@ -439,11 +469,22 @@ func (r *DPAReconciler) getAzureRegistryEnvVars(bsl *velerov1.BackupStorageLocat
 	}
 
 	// parse the secret and get azure storage account key
-	AzureStorageKey, err := r.parseAzureSecret(secret, secretKey)
+	azcreds, err := r.parseAzureSecret(secret, secretKey)
 	if err != nil {
 		r.Log.Info(fmt.Sprintf("Error parsing provider secret %s for backupstoragelocation %s/%s", secretName, bsl.Namespace, bsl.Name))
 		return nil, err
 	}
+	if len(bsl.Spec.Config["storageAccountKeyEnvVar"]) != 0 {
+		if azcreds.strorageAccountKey == "" {
+			r.Log.Info("Expecting storageAccountKeyEnvVar value set present in the credentials")
+			return nil, errors.New("no strorageAccountKey value present in credentials file")
+		}
+	} else {
+		r.Log.Info("Checking for service principal credentials")
+		if len(azcreds.subscriptionID) == 0 && len(azcreds.tenantID) == 0 && len(azcreds.clientID) == 0 && len(azcreds.clientSecret) == 0 && len(azcreds.resourceGroup) == 0 {
```

Comment:
Do we not need `strorageAccountKey` value as well for Service principal ?

Reply:
we need `strorageAccountKey`. Hence checking for the one in bslspec. If the config parameter for `storageAccountKeyEnvVar` is not present, then service principal creds needs to be validated

Reply:
Irresepctive of whatever method we need the `strorageAccountKey` in credential file.

Reply:
just a NIT for this... when we have this many conditionals on an if statement looks better if you do:
```
if len(azcreds.subscriptionID) == 0 &&
    len(azcreds.tenantID) == 0 && 
    len(foo)
   ...
```
```